frontend/src/lib/utils.ts

```diff
@@ -203,3 +207,49 @@ export function labelVisibilityProtoValueToEnum(
       return LabelVisibilityOptions.Visible
   }
 }
+
+export function isAppInLoadingState(): boolean {
```

Comment:
This is great advice, I must admit, I have not read `App.tsx` code thoroughly, after this advice I did and the following changes are introduced:
* instead of checking visual state of the app I'm using `scriptFinishedHandlers`, which are always run when `handleScriptFinished` is run, I've checked it in multiple cases and it's always consistent with `isAppInLoadingState`, thus I've removed `isAppInLoadingState` and related logic